### DISCUSSION TITLE
Is there a way to completely disable preselect?

### DISCUSSION THREAD

**Discussion by yzhou216** [minad/corfu#487] [Q&A] [answered]
Is it possible for the automatic popup to still show, however, set it so that it never selects anything unless I manually scroll with arrow keys or `C-n` and `C-p` with Evil mode. I tried setting `corfu-preselect` to `nil` and that didn't work.

**Comment by minad** [comment]
Please check out the other customization values of `corfu-preselect`, in particular `prompt`.

**Comment by yzhou216** [comment]
Thanks for the reply, I tried out all four options, and I don't think any of them is the behavior I want. With  `corfu-preselect` set to `prompt`, `RET` still doesn't function as a "normal" return key (inserting a new line), instead, it closes the popup.

**Reply by minad** [reply]
You might be looking for this: https://github.com/minad/corfu?tab=readme-ov-file#tab-only-completion

**Reply by yzhou216** [reply]
I still want `RET` to select the completion, meaning if I move down on the popup and press return, I want it to still select what's highlighted. However, I want there's no initial preselect, meaning if I don't manually move down the popup, `RET` shouldn't do anything but inserting a new line, and as soon as I move down (or up) in the pop, `RET` functions as select. I also want to leave `TAB` untouched, since I use it for snippet expansion already.

**Reply by minad** [reply, marked as answer]
This precise mode of operation is not supported by Corfu, intentionally so, since Corfu tries to operate uniformly. What you describe for RET is non-uniform, inserting a newline or selecting a candidate. However you can easily implement this yourself. Set `corfu-preselect=prompt` and implement custom RET/TAB commands.
